hydra-node/test/Hydra/Chain/Direct/ContractSpec.hs

```diff
@@ -125,13 +125,14 @@ prop_verifySnapshotSignatures =
           snapshotNumber = toInteger $ number snapshot
        in verifySnapshotSignature parties snapshotNumber signatures
 
-propMutation :: (CardanoTx, Utxo) -> ((CardanoTx, Utxo) -> Gen (Text, Mutation)) -> Property
+propMutation :: (CardanoTx, Utxo) -> ((CardanoTx, Utxo) -> Gen (SomeMutationLabel, Mutation)) -> Property
 propMutation (tx, utxo) genMutation =
-  forAll (genMutation (tx, utxo)) $ \(_label, mutation) ->
+  forAll (genMutation (tx, utxo)) $ \(SomeMutationLabel lbl, mutation) ->
     (tx, utxo)
       & applyMutation mutation
       & propTransactionDoesNotValidate
-      & withMaxSuccess 1000
+      & genericCoverTable @ 'UniformCoverage [lbl]
```

Comment:
I don't understand what `[lbl]` is doing here?

Reply:
Is it just do indicate the `lbl` type as a proxy to `gCoverTableRequirements`?

Reply:
It counts the occurrence of that label in the property execution. A`Property` is intrinsically stateful and quickcheck records things during a property execution like label occurences in order to (a) determine whether the execution meets the coverage requirements and (b) provide a nice report at the end.

Reply:
I took some shortcut with `genericCoverTable` which is also calling `tabulate` and it's maybe the source of the confusion here. So, not only does this declare the table to cover, but it also record the occurrence of some of the cases we are after.

Reply:
Did this because I found the syntax `coverTable + tabulate` needlessly heavy. 99% of the time, they are called one after the other.